inst/include/cleancall.h

```diff
@@ -0,0 +1,34 @@
+#ifndef R_CLEANCALL_H
+#define R_CLEANCALL_H
+
+
+static R_INLINE SEXP r_with_cleanup_context(SEXP (*fn)(void* data),
+                                            void* data) {
+  static SEXP (*fun)(SEXP (*func)(void*), void *data) = NULL;
+  if (fun == NULL) {
```

Comment:
Can you use `fn` instead of `fun` and `func` which is how we abbreviate function in our style guide?

Reply:
Yeah, these are not the best names, but we have three of them here, so three names are needed.

Reply:
I think we only need the binding names to be different. You can either leave the names in the rest of the definition, or reuse `fn` which has a congruent type. I suggest:

```c
static SEXP (*cleancall_ptr)(SEXP (*fn)(void*), void *data) = NULL;
```